monitoring: customize prometheus rule alerts (#9503)
rook/rook: Documentation/ceph-cluster-crd.md

    #externalMgrEndpoints:
      #- ip: 192.168.39.182
    #externalMgrPrometheusPort: 9283
    # prometheus rule alerts values for overriding default prometheus rules values
    # optional: specific alerts can be disabled by setting disabled field to true 
    #alertRuleOverrides:
    #  CephNodeDown:
    #    disabled: true
    #  CephMgrIsAbsent:
    #    for: 1m

>     #    severityLevel: custom-severityLevel

Reviewer: Do these need to be from a specific set such as warning, error, critical? Or can custom levels be defined somehow?

Author: I wasn't familiar with the ceph severity_level UI requirment explained [here](https://github.com/ceph/ceph-mixins/issues/54#issuecomment-488950655).

Now, if there is custom severity label that is not one of the accepted levels:  [warning, error, critical], prometheus template will be failed.

We can decide that instead of failing the templating in that case, we can just use the default value instead of the wrong custom value and log the reason for using the default value.

Please let me know which way you prefer to implement.

>     #    severityLevel: <custom-severityLevel> #<custom-severityLevel> must be one of the next levels: warning, critical, error

Reviewer: how about we just use one of the examples values here?
```suggestion
    #    severityLevel: warning # must be warning, critical, or error
```

Author: Sure, make sense.

Author: Regarding the suggestion about enum on both of the fields,  `severity` and `severityLevel` are not exactly the same. `severity` is a label, so we can use a custom severity label in Prometheus. It might help if you would like to direct different rook alerts to different slack channels.

For example,
You can give some of the alerts the  `severity: rook-warning`, so all the alerts with this label can be sent to a different channel that is separated from alerts with the  `severity: warning`.

I believe that in `severity` field, we should let the user decide which severity label to use.
